src/main/java/de/tum/in/www1/artemis/service/metis/conversation/GroupChatService.java

```diff
@@ -0,0 +1,120 @@
+
+    private static final String GROUP_NAME_REGEX = "^[a-z0-9-]{0,20}$";
+
+    private final UserRepository userRepository;
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final ConversationService conversationService;
+
+    private final GroupChatRepository groupChatRepository;
+
+    public GroupChatService(UserRepository userRepository, ConversationParticipantRepository conversationParticipantRepository, ConversationService conversationService,
+            GroupChatRepository groupChatRepository) {
+        this.userRepository = userRepository;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationService = conversationService;
+        this.groupChatRepository = groupChatRepository;
+    }
+
+    /**
+     * Gets a group chat by its id from the database or throws an exception if it does not exist
+     *
+     * @param groupChatId the id of the group chat
+     * @return the group chat
+     */
+    public GroupChat getGroupChatOrThrow(Long groupChatId) {
+        return groupChatRepository.findById(groupChatId)
+                .orElseThrow(() -> new BadRequestAlertException("GroupChat with id " + groupChatId + " does not exist", GROUP_CHAT_ENTITY_NAME, "idnotfound"));
+    }
+
+    /**
+     * Creates a new group chat
+     *
+     * @param course          the course the group chat is created in
+     * @param startingMembers the users that are initially part of the group chat
+     * @return the newly created group chat
+     */
+    public GroupChat startGroupChat(Course course, Set<User> startingMembers) {
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        var groupChat = new GroupChat();
+        groupChat.setCourse(course);
+        groupChat.setCreator(requestingUser);
+        var savedGroupChat = groupChatRepository.save(groupChat);
+        var participantsToCreate = startingMembers.stream().map(user -> createChatParticipant(user, groupChat)).toList();
+        conversationParticipantRepository.saveAll(participantsToCreate);
+        return savedGroupChat;
+    }
+
+    @NotNull
+    private ConversationParticipant createChatParticipant(User user, GroupChat groupChat) {
+        var participant = new ConversationParticipant();
+        participant.setUser(user);
+        participant.setConversation(groupChat);
+        // makes no sense for group chats
+        participant.setIsAdmin(false);
+        participant.setIsFavorite(false);
+        participant.setIsHidden(false);
+        return participant;
+    }
+
+    /**
+     * Updates the group chat with the given id
+     *
+     * @param groupChatId  the id of the group chat to update
+     * @param groupChatDTO the DTO containing the new group chat data
+     * @return the updated group chat
+     */
+    public GroupChat updateGroupChat(Long groupChatId, GroupChatDTO groupChatDTO) {
```

Comment:
This logic is only related to storing an updated group chat? Could be directly done in the Resource class.

Reply:
I would like to keep the update logic in the service as with the other types.